sopel/modules/safety.py

```diff
@@ -7,75 +7,112 @@
+    | known\\_good | sopel.chat,dftba.net | List of "known good" domains or regexes to ignore. This can save VT API calls. |
     | vt\\_api\\_key | 0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef | Optional VirusTotal API key to improve malicious URL detection |
+    | domain\\_blocklist\\_url | https://example.com/bad-hosts.txt | Optional hosts-file formatted domain blocklist to use instead of StevenBlack's. |
     """
-    config.define_section('safety', SafetySection)
-    config.safety.configure_setting(
-        'enabled_by_default',
-        "Enable URL safety in channels that don't specifically disable it?",
+    settings.define_section("safety", SafetySection)
+    settings.safety.configure_setting(
+        "default_mode",
+        (
+            "Which mode should be used in channels that haven't specifically set one?"
+            "\n({})".format("/".join(SAFETY_MODES))
+        ),
+        default="on",
     )
-    config.safety.configure_setting(
+    settings.safety.configure_setting(
         'known_good',
-        'Enter any domains to whitelist',
+        "Enter any domains to allowlist",
     )
-    config.safety.configure_setting(
+    settings.safety.configure_setting(
         'vt_api_key',
         "Optionally, enter a VirusTotal API key to improve malicious URL "
-        "protection.\nOtherwise, only the StevenBlack list will be used."
+        "protection.\nOtherwise, only the configured hosts list will be used.",
+    )
+    settings.safety.configure_setting(
+        "domain_blocklist_url",
+        "Optionally, provide the URL for a hosts-file formatted domain "
+        "blocklist to use instead of StevenBlack's.",
     )
 
 
-def setup(bot):
-    bot.config.define_section('safety', SafetySection)
+def setup(bot: Sopel):
+    bot.settings.define_section("safety", SafetySection)
+
+    if bot.settings.safety.default_mode is None:
+        bot.settings.safety.default_mode = "on"
+    # migrate from enabled_by_default to default_mode. remove in v8.1 or v9
+    if not bot.settings.safety.enabled_by_default:
+        bot.settings.safety.default_mode = "off"
+        LOGGER.info(
+            "config: enabled_by_default is deprecated, please use default_mode=off",
+        )
 
     if 'safety_cache' not in bot.memory:
         bot.memory['safety_cache'] = tools.SopelMemory()
     if 'safety_cache_lock' not in bot.memory:
         bot.memory['safety_cache_lock'] = threading.Lock()
-    for item in bot.config.safety.known_good:
+    for item in bot.settings.safety.known_good:
         known_good.append(re.compile(item, re.I))
 
-    old_file = os.path.join(bot.config.homedir, 'malwaredomains.txt')
+    update_local_cache(bot, init=True)
+
+
+def update_local_cache(bot: Sopel, init: bool = False):
+    """Download the current malware domain list and load it into memory.
+
+    :param init: Load the file even if it's unchanged
+    """
+
```

Comment:
```suggestion
```

One day I will get around to installing/configuring/whatever the flake8 plugin/rule that flags this as an error.